backend/app/src/channel/channel.service.ts

```
@@ -233,30 +319,67 @@ export class ChannelService {
       if (error.code === 'P2002') {
         return 'alreadyUsed' + error.meta.target[0];
       }
+      if (error == 'Error: WrongData') {
+        return 'WrongData';
+      }
       return null;
     }
   }
 
+  async getIsInvitedInAChannel(userId: string, channelId: string) {
+    try {
+      const isInvited = await this.prisma.channel.findUnique({
+        where: {
+          id: channelId,
+        },
+        select: {
+          invites: {
+            select: {
+              id: true,
+            },
+          },
+        },
+      });
+      return isInvited.invites.find((value) => {
+        return value.id === userId;
+      })
+        ? true
+        : false;
+    } catch (error) {
+      return error;
+    }
+  }
+
   async joinChannelWS(
     channelDto: JoinChannelDto,
     userId: string,
     clientSocket: Socket,
   ) {
     try {
-      /* Get the channel's password if the type is protected */
-      const channel: { type: ChannelType; passwordHash: string } =
-        await this.prisma.channel.findFirst({
-          where: {
-            id: channelDto.id,
-            type: ChannelType.PROTECTED,
-          },
-          select: {
-            type: true,
-            passwordHash: true,
-          },
-        });
+      /* Check the password is provided in the DTO for protected chan) */
+
+      /** If private channel, check the invitation to the channel */
+      if (channelDto.type === ChannelType.PRIVATE) {
+        const isInvited = await this.getIsInvitedInAChannel(
```

case where getIsInvitedInAChannel returns an error is not handled I think?